src/tests/decider-distro-tests-rs/tests/test.rs

```diff
@@ -119,3 +533,1006 @@ async fn test_decider_installed() {
+///    In this test we don't have any updates or mailbox messages.
+///
+/// 7. After creation, *Worker Spell* downloads the app from IPFS and deploys it on the worker
+///    Here, as a part of testing the basic flow, we check that the app was deployed correctly:
+///    - `srv.list` on the worker returns the `worker-spell` spell and the `url-downloader` service
+///
+/// 8. *Decider* polls updates for already existing deals.
+///    Check that *Decider* doesn't try to find updates for the new deal.
+///
+/// NOTE: This test REQUIRES an IPFS node to be up and have the url-downloader app uploaded.
+/// TODO: provide the app in the tests resources
+/// TODO: checks that `errors` are empty
+///
+#[tokio::test]
+async fn test_deploy_a_deal_single() {
+    const DEAL_ID: &'static str = DEAL_IDS[0];
+    const BLOCK: u32 = 32;
+
+    let mut server = run_test_server_predefined(async move |method, params| {
+        match method.as_str() {
+            "eth_blockNumber" => {
+                json!("0x10")
+            }
+            "eth_getLogs" => {
+                let log = serde_json::from_value::<LogsReq>(params[0].clone()).unwrap();
+                Value::Array(vec![TestApp::log_test_app1(
+                    DEAL_ID,
+                    BLOCK,
+                    log.topics[1].as_str(),
+                )])
+            }
+            "eth_sendRawTransaction" => {
+                // how to check correctness of the subnet registration?
+                json!("0x55bfec4a4400ca0b09e075e2b517041cd78b10021c51726cb73bcba52213fa05")
+            }
+            "eth_getTransactionCount" => json!("0x1"),
+            "eth_gasPrice" => json!("0x3b9aca07"),
+            _ => panic!("unexpected method: {}", method),
+        }
+    });
+
+    let url = server.url.clone();
+
+    let distro = make_distro_with_api(url);
+    let swarms = make_swarms_with_cfg(1, move |mut cfg| {
+        cfg.enabled_system_services = vec!["aqua-ipfs".to_string()];
+        cfg.extend_system_services = vec![distro.clone()];
+        cfg
+    })
+    .await;
+
+    let mut client = ConnectedClient::connect_with_keypair(
+        swarms[0].multiaddr.clone(),
+        Some(swarms[0].management_keypair.clone()),
+    )
+    .await
+    .unwrap();
+
+    update_decider_script_for_tests(&mut client, swarms[0].tmp_dir.clone()).await;
+    update_config(&mut client, &oneshot_config()).await.unwrap();
+
+    wait_decider(&mut client).await;
+
+    let mut result = execute(
+        &mut client,
+        r#"
+            (seq
+                (call relay ("decider" "get_u32") ["counter"] counter)
+                (call relay ("decider" "list_get_strings") ["joined_deals"] deals)
+            )
+        "#,
+        "counter deals",
+        hashmap! {},
+    )
+    .await
+    .unwrap();
+
+    // Check that decider was really run
+    let counter = serde_json::from_value::<U32Value>(result.remove(0)).unwrap();
```

Comment:
i'd move `result.remove(0)` to a separate variable with a telling name. `result[0]` is hard to follow